src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +36,15 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        List<Integer> modifiedList = IntStream.range(0, numbers.size())
+                .mapToObj(i -> i % 2 == 0 ? numbers.get(i) : numbers.get(i) - 1)
+                .collect(Collectors.toList());
+
+        return modifiedList.stream()
+                .filter(num -> num % 2 != 0)
+                .mapToDouble(Integer::doubleValue)
+                .average()
+                .orElseThrow(NoSuchElementException::new);
```

Comment:
1) There is no sense to collect stream to open new one again.
2) Too many redundant commands
```suggestion
        List<Integer> modifiedList = IntStream.range(0, numbers.size())
                .map(i -> i % 2 == 0 ? numbers.get(i) : numbers.get(i) - 1)
                .filter(num -> num % 2 != 0)
                .average()
                .orElseThrow(NoSuchElementException::new);
```